File: connectx-backend/src/main/java/com/connectx/auth/dto/ForgotPasswordRequestDto.java
package com.connectx.auth.dto;

import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;

public class ForgotPasswordRequestDto {

    @NotBlank(message = "Email is required")
    @Email(message = "Invalid email format")
    private String email;

    public ForgotPasswordRequestDto() {}

    public ForgotPasswordRequestDto(String email) {
        this.email = email;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }
}

File: connectx-android/app/src/main/java/com/connectx/app/data/remote/auth/model/ForgotPasswordRequestDto.kt
package com.connectx.app.data.remote.auth.model

import kotlinx.serialization.Serializable

/**
 * Mirrors `com.connectx.auth.dto.ForgotPasswordRequestDto` exactly. Backend
 * @NotBlank/@Email -- required. Email-based recovery only, matching
 * ConnectX's actual identity model (username for discovery/login, email for
 * account recovery) -- no phone number field exists on the backend, so none
 * is modeled here.
 */
@Serializable
data class ForgotPasswordRequestDto(
    val email: String
)
